app/src/main/java/de/tum/in/tumcampusapp/activities/SurveyActivity.java

```diff
@@ -440,21 +452,76 @@ protected void onPostExecute(Void v) {
                         }.execute();
 
                     } else {
-                        // Without newly created questions
-                        setUpResponseTab();
+                        setUpResponseTab(); // setup responses tab Without possible fresh answeres.
                     }
                 }
             }
         });
     }
-    private void restartActivity() {
-        Intent i = getIntent();
-        i.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
-        finish();
-        startActivity(i);
+
+
+    /**
+     * Help function for plausibility check of the question form
+     *
+     * @return true if everything was entered correctly else snackbar for requesting to complete questions.
+     */
+    private boolean checkSurveyData() {
+        boolean done = true;
+        for (int i = 0; i < aSpinner1.getSelectedItemPosition() + 1; i++) { // Iterates on each questionEditText
+            EditText v = (EditText) questionsLayout.findViewWithTag("question" + (i + 1));
+            if (!v.getText().toString().isEmpty()) { // if textfield is not empty, add to questions
+                questions.add(v.getText().toString());
+            } else { // else plausibility check failed
+                done = false;
+                questions.clear();
+                break;
+            }
+        }
+        if (done) { // if plausibility passed, then save selected faculties as they will be needed upon submitting question
+            chosenFaculties = getSelectedFaculties(selectedFaculties);
+            return true;
+        } else { // else notify the user with a snackbar to complete the question form
+            Snackbar.make(findViewById(R.id.drawer_layout), getResources().getString(R.string.complete_question_form), Snackbar.LENGTH_LONG).show();
+        }
+        return false;
+    }
+
+    /**
+     * Help function for clearing data and layout entries after submitting questions
+     */
+    private void clearData() {
+        selectedFaculties.clear();
+        chosenFaculties = "";
+        questions.clear();
+        for (int i = 0; i < checked.length; i++) { // uncheck selected faculties
+            checked[i] = false;
+        }
+        aSpinner1.setSelection(0); //
+    }
+
+    //get selected faculties from spinner
+
+    /**
+     * Help function for getting all faculties as a string from the spinner
+     * @param arrayList: selected faculties in spinner as arraylist
+     * @return: selected faculties as String
+     */
+    public String getSelectedFaculties(ArrayList<String> arrayList) {
```

Comment:
Can you generalize this method and put it into the Utils?

PHP has a very similar function: http://php.net/manual/de/function.implode.php
Bonus points if use something from here: http://stackoverflow.com/questions/11248119/java-equivalent-of-phps-implode-array-filter-array